app/src/main/java/de/westnordost/streetcomplete/quests/incline_direction/AddBicycleIncline.kt

```diff
@@ -0,0 +1,62 @@
+package de.westnordost.streetcomplete.quests.incline_direction
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementPolylinesGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmQuest
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.incline_direction.InclineDirection.UP
+import de.westnordost.streetcomplete.quests.incline_direction.InclineDirection.UP_REVERSED
+import de.westnordost.streetcomplete.util.math.measuredLength
+
+class AddBicycleIncline : OsmElementQuestType<InclineDirection> {
+
+    private val tagFilter by lazy { """
+        ways with mtb:scale:uphill
+         and highway ~ footway|cycleway|path|bridleway|track
+         and (!indoor or indoor = no)
+         and area != yes
+         and access !~ private|no
+         and !incline
+    """.toElementFilterExpression() }
+
+    override val changesetComment = "Specify which way leads up (where mtb:scale:uphill is present)"
+    override val wikiLink = "Key:incline"
+    override val icon = R.drawable.ic_quest_bicycle_incline
+    override val achievements = listOf(BICYCLIST)
+    override val hasMarkersAtEnds = false
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+        // sadly for very long ways shape may be complex and it may be confusing which answer should be given
+        // once multiple quest markers are appearing it becomes completely unclear
+        // see for example https://www.openstreetmap.org/way/437205914
+        return mapData
+            .filter { tagFilter.matches(it) }
+            .filter {
+                val geometry = mapData.getGeometry(it.type, it.id) as? ElementPolylinesGeometry
+                val overlyLong = geometry?.polylines?.filter { it.measuredLength() > OsmQuest.minLengthForMultiMarkers(false) }
```

Comment:
This is arbitrary. Please just specify a constant here instead of referring to some unrelated function. Also, 800+ meters sounds a bit long for choosing to not display this quest for long ways.

Also, why can the user not simply split up the way in such a case?

Reply:
That is actually a related function: if there are multiple markers on a windy paths then it becomes utterly unclear which direction should be actually applied.

![screen14](https://user-images.githubusercontent.com/899988/190648486-92455f6d-db6b-4331-bd3a-a7c3a766d421.png)

> Also, 800+ meters sounds a bit long for choosing to not display this quest for long ways.

Though I guess that some small distance can be also used, with assumption that it will always have a single marker

>Also, why can the user not simply split up the way in such a case?

It is not entirely obvious (at least I have not thought about it) and in many cases would be done solely as workaround for SC interface so probably should not be encouraged

Reply:
Why would it be done as a workaround to SC interface?

Reply:
In case where incline changes it should be split, and if parts of it are small enough this stops being relevant.

In case where way has continuous this case primary motivation for splitting way would be to get its direction displayed clearly in SC (so: workaround to SC interface)

Reply:
> In case where incline changes it should be split, and if parts of it are small enough this stops being relevant.

Well you have the same for any length of way.